### PR TITLE
Account lookup failures, account and client limits, options reload.

### PR DESCRIPTION
**Pull request by derekcollison**
Changed account lookup and validation failures to be more understandable by users.
Changed limits to be -1 for unlimited to match jwt pkg.

The limits changed exposed problems with options holding real objects causing issues with reload tests under race mode.
Longer term this code should be reworked such that options only hold config data, not real structs, etc.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/reload.go [hunk 1]
```diff
@@ -637,6 +650,10 @@ func (s *Server) diffOptions(newOpts *Options) ([]option, error) {
 			diffOpts = append(diffOpts, &clientAdvertiseOption{newValue: cliAdv})
 		case "accounts":
 			diffOpts = append(diffOpts, &accountsOption{})
+		case "trustedkeys":
+			fmt.Printf("newVal is %+v\n", newValue)
```
Comment (kozlovic) [inline server/reload.go]: Remove

File: main.go [hunk 2]
```diff
@@ -87,16 +90,16 @@ func main() {
 		fs.Usage,
 		server.PrintTLSHelpAndDie)
 	if err != nil {
-		server.PrintAndDie(fmt.Sprintf("ERR: %s", err))
+		server.PrintAndDie(fmt.Sprintf("%s: %s", exe, err))
 	} else if opts.CheckConfig {
-		fmt.Fprintf(os.Stderr, "configuration file %s is valid\n", opts.ConfigFile)
+		fmt.Fprintf(os.Stderr, "nats-server: configuration file %s is valid\n", opts.ConfigFile)
```
Comment (kozlovic) [inline main.go]: use `exe` here then?

File: server/reload.go [hunk 3]
```diff
@@ -717,33 +734,39 @@ func (s *Server) applyOptions(opts []option) {
...
-			// sublist to the new account object before registering it
-			// in s.accounts.
+			// sublist and client map to the new account.
 			acc.mu.RLock()
 			sl := acc.sl
+			clients := acc.clients
 			acc.mu.RUnlock()
 			newAcc.mu.Lock()
 			newAcc.sl = sl
+			if len(clients) > 0 {
```
Comment (kozlovic) [inline server/reload.go]: Technically, a client could be removed from the account after just just capture the pointer, so ranging down below would still be a problem, no?

File: server/reload.go (hunk 1 above)
Comment (derekcollison) [reply server/reload.go]: Will fix.

File: main.go (hunk 2 above)
Comment (derekcollison) [reply main.go]: yes

File: server/reload.go (hunk 3 above)
Comment (derekcollison) [reply server/reload.go]: Not understanding. Maybe GH to discuss.

**Comment by derekcollison** [comment]
I did a rebase from master which pulled in Pires stuff..

**Pull request merged**